backend/controllers/timeLogs.js

```diff
@@ -0,0 +1,161 @@
+    return 'Description must be over 5 characters.'
+  }
+  return null
+}
+
+
+const fetchFromDb = async (studentNumber) => {
+  try {
+    const raw_logs = await db.TimeLog.findAll({
+      where: { student_number: studentNumber },
+      attributes: [
+        ['id', 'id'],
+        ['date', 'date'],
+        ['minutes', 'minutes'],
+        ['student_number', 'studentNumber'],
+        'description',
+        [Sequelize.literal('sprint.sprint'), 'sprint'],
+        [Sequelize.literal('group_id'), 'groupId'],
+      ],
+      include: [
+        {
+          model: db.Sprint,
+          as: 'sprint',
+          attributes: [],
+        },
+        {
+          model: db.Tag,
+          as: 'tags',
+          attributes: ['title'],
+          through: { attributes: [] }
+        }
+      ],
+      raw: true,
+    })
+
+    const timeLogMap = new Map()
+    raw_logs.forEach(log => {
+      const { id, studentNumber, sprint, date, minutes, description, groupId, ...rest } = log
+      const formattedDate = new Date(date).toISOString().slice(0, 10)
+      const timeLog = timeLogMap.get(id) || { id, studentNumber, sprint, date: formattedDate, minutes, description, groupId, tags: [] }
+      const tag = rest['tags.title']
+      if (tag) {
+        timeLog.tags.push(tag)
+      }
+      timeLogMap.set(id, timeLog)
+    })
+    return Array.from(timeLogMap.values())
+  } catch (error) {
+    console.error('Error fetching time logs:', error)
+    throw new Error('Error fetching time logs.')
+  }
+}
+
+timeLogsRouter.get('/', checkLogin, async (req, res) => {
+  try {
+    const user_id = req.user.id
+    const timeLogs = await fetchFromDb(user_id)
+    return res.status(200).json(timeLogs)
+  } catch (error) {
+    return res.status(500).json({ error: 'Error fetching time logs.' })
+  }
+})
+
+timeLogsRouter.post('/', checkLogin, async (req, res) => {
+  //req.body.tags = ['Frontend', 'Backend', 'React'] // Mock data, remove
+  const { sprint, date, minutes, description, tags, groupId } = req.body
+  const studentNumber = req.user.id
+  const newLog = {
+    studentNumber,
+    sprint,
+    date,
+    minutes,
+    description,
+    tags,
+    groupId,
+  }
+
+  const error = validateTimeEntry(newLog)
+  if (error) {
+    console.error('error:', error)
+    return res.status(400).json({ error })
+  }
+
+  try {
+    const sprintRecord = await db.Sprint.findOne({
+      where: {
+        sprint: sprint,
+        group_id: groupId
+      }
+    })
+    if (!sprintRecord) {
+      return res.status(404).json({ error: 'Sprint not found.' })
+    }
+
+    const sprintId = sprintRecord.id
+
+    const timeLog = await db.TimeLog.create({
+      student_number: studentNumber,
+      sprint_id: sprintId,
+      date: date,
+      minutes: minutes,
+      description: description,
+    })
+
+    // this simply ignores any tags that don't exist in the database
+    // error could be raised if the tags are not found, but is it necessary?
```

Comment:
TODO

Reply:
removed comment. I think this would be unnecessary check